tests/countries/test_ivory_coast.py

```diff
@@ -0,0 +1,184 @@
+
+    def test_assumption_day(self):
+        self.assertHolidayName(
+            "Fête de l’Assomption", (f"{year}-08-15" for year in range(1997, 2050))
+        )
+
+    def test_all_saints_day(self):
+        self.assertHolidayName(
+            "Fête de la Toussaint", (f"{year}-11-01" for year in range(1997, 2050))
+        )
+
+    def test_christmas_day(self):
+        self.assertHolidayName("Fête de Noël", (f"{year}-12-25" for year in range(1997, 2050)))
+        self.assertHolidayName("Lendemain de la Fête de Noël", "2011-12-26")
+
+    def test_eid_al_fitr(self):
+        name = "Fête de fin du Ramadan"
+        self.assertHolidayName(
+            name,
+            "2021-05-13",
+            "2022-05-02",
+            "2023-04-21",
+            "2024-04-10",
+            "2025-03-30",
+        )
+        self.assertHolidayName(name, range(1997, 2050))
+        self.assertHolidayName("Lendemain de la Fête de fin du Ramadan", "2017-06-26")
+
+    def test_eid_al_adha(self):
+        name = "Fête de la Tabaski"
+        self.assertHolidayName(
+            name,
+            "2021-07-20",
+            "2022-07-09",
+            "2023-06-28",
+            "2024-06-16",
+            "2025-06-06",
+        )
+        self.assertHolidayName(name, range(1997, 2050))
+        self.assertHolidayName("Lendemain de la Fête de la Tabaski", "2019-08-12")
+
+    def test_mawlid(self):
+        name = "Lendemain de l’Anniversaire de la Naissance du Prophète Mahomet"
+        self.assertHolidayName(
+            name,
+            "2021-10-18",
+            "2022-10-08",
+            "2023-09-27",
+            "2024-09-15",
+            "2025-09-04",
+        )
+        self.assertHolidayName(name, range(1997, 2050))
+
+    def test_laylat_al_qadr(self):
+        name = "Lendemain de la Nuit du Destin"
+        self.assertHolidayName(
+            name,
+            "2021-05-09",
+            "2022-04-28",
+            "2023-04-18",
+            "2024-04-06",
+            "2025-03-27",
+        )
+        self.assertHolidayName(name, range(1997, 2050))
+
+    def test_anniversary_of_death_of_first_president(self):
+        name = "Anniversaire du décès du Président Felix Houphouet-Boigny"
+        self.assertHolidayName(
+            name,
+            "1997-12-07",
+            "1998-12-07",
+            "1999-12-07",
+            "2000-12-07",
+        )
+        self.assertNoHolidayName(name, range(2001, 2050))
+
+    def test_static_holidays(self):
+        self.assertHolidayName(
+            "Jour férié pour la préparation de l'élection présidentielle", "2010-10-29"
+        )
+        self.assertNoHolidayName(
+            "Jour férié pour la préparation de l'élection présidentielle", "2011-10-29"
+        )
+
+        self.assertHolidayName("Victoire à la Coupe d’Afrique des Nations 2024", "2024-02-12")
+        self.assertNoHolidayName("Victoire à la Coupe d’Afrique des Nations 2024", "2025-02-12")
```

Comment:
```suggestion
    def test_special_holidays(self):
        self.assertHoliday(
            "2010-10-29",
            "2024-02-12",
        )
```
And this should be placed after `test_no_holidays()`.

And at the end, we need localization tests. You can look at the example in any country with localization.